guides/dsl.md

```diff
@@ -0,0 +1,316 @@
+# Eve JavaScript/TypeScript DSL
+
+The forthcoming v0.3 release of Eve supports a domain specific language (DSL) that interacts with the Eve runtime, allowing you to intermix Javascript and Eve code. There are several benefits to using the DSL:
+
+1) The syntax is native Javascript, so if you know JS you can write in the DSL
+2) You can interact with Javascript functions and libraries
+3) The DSL can be used "a la carte", so you can use as much of Eve as you need to for your project. Thus you can write an entire application in Eve, just use it as a datastore, or anywhere between. This makes it easy to integrate Eve with an existing Javascript application.
+4) You can easily import data into Eve. If you can get your data in Javascript, you can use it in Eve.
+
+This DSL guide is for users who are already familiar with Eve semantics. For those new to Eve, we'll have more fundamental guides released in the near future.
+
+## Summary of DSL 
+
+- find a record: `find("person", {salary})` <-> `[#person salary]`
+- bind/commit a record: `record("person", {salary})` <-> `[#person salary]`
+- not: `not(() => person.salary)` <-> `not(person.salary)`
+- choose: `choose(() => { person.salary; return 1; }, () => 0)` <-> `if person.salary then 1 else 0`
+- union: `union(() => { person.salary; return salary; }, () => person.wage; return wage)` <-> `if person.salary then salary if person.wage then wage`
```

Comment:
For the DSL case you can simplify this to:

`union(() => person.salary, () => person.wage)` :)